LoopKit/DeviceAlert.swift

```diff
@@ -79,15 +79,34 @@ public struct DeviceAlert {
     /// An alert's "identifier" is a tuple of `managerIdentifier` and `alertIdentifier`.  It's purpose is to uniquely identify an alert so we can
     /// find which device issued it, and send acknowledgment of that alert to the proper device manager.
     public let identifier: Identifier
-        
-    public init(identifier: Identifier, foregroundContent: Content?, backgroundContent: Content?, trigger: Trigger) {
+
+    // Name of the sound file suitable for using to play back
+    public typealias SoundName = String
+    public let soundName: SoundName?
+    
+    public init(identifier: Identifier, foregroundContent: Content?, backgroundContent: Content?, trigger: Trigger, soundName: SoundName? = nil) {
         self.identifier = identifier
         self.foregroundContent = foregroundContent
         self.backgroundContent = backgroundContent
         self.trigger = trigger
+        self.soundName = soundName
     }
 }
 
+// Special name to mean "vibrate"
+public extension DeviceAlert.SoundName {
+    static let vibrate: DeviceAlert.SoundName = "__vibrate__"
+    static let silence: DeviceAlert.SoundName = "__silence__"
+}
```

Comment:
Given that you're extending the type with public static members, it would be preferable to define `SoundName` as `struct SoundName { var rawValue: String }` to give the type its own namespace.